backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java

```diff
@@ -634,15 +642,23 @@ public void validateDiffMergeRequest(
       if (conflictLocationMap.isEmpty()) {
         final String rootSha = setBlobs(writeSession, blobContainerList, new FileHasher());
         long timeCreated = new Date().getTime();
-        List<String> parentSHAs = Arrays.asList(request.getCommitShaA(), request.getCommitShaB());
+        List<String> parentSHAs = Arrays.asList(internalCommitA.getCommit_hash(), internalCommitB.getCommit_hash());
         List<CommitEntity> parentCommits = Arrays.asList(internalCommitA, internalCommitB);
         String mergeMessage = request.getContent().getMessage();
         if (mergeMessage.isEmpty()) {
-          mergeMessage =
-              "Merge "
-                  + request.getCommitShaA().substring(0, 7)
-                  + " into "
-                  + request.getCommitShaB().substring(0, 7);
+          if (branchOrCommitA.length() > 7) {
```

Comment:
if it is a branch name then do not truncate the string

Reply:
if youtruncate the commit sha on top then you can just. have the code you have in else

Reply:
Fixed